### DISCUSSION TITLE
Routing for pagination and articles

### DISCUSSION THREAD

**Discussion by popeating** [vercel/next.js#24593] [Help]
I'm converting a website/magazine with about 1000 articles, the actual structure is as follow:

https://mysite.com/ -> homepage
https://mysite.com/category/[pagenumber] -> archive page for category
https://mysite.com/category/subcategory/[pagenumber] -> archive for subcategory (luckly im limited to 1 subcategory)
https://mysite.com/category/article -> single article for category
https://mysite.com/category/subcategory/article -> single article for subcategory

at the moment i have no problem at generating the page, also the single article for category with no subcategory (i made a [category]/[slug].js path)
but now im trying to code the archive and i can't get out of it:
i can't use a [category]/[pagenumber].js since it conflicts with the previous one, also the first page has no number, how to handle it?
Also, for the pages with subcategory, how can i still use the [slug.js] without duplicating it in [category]/[subcategory]/[slug].js?

i know i can change the structure, but at the moment the site is well indexed (and i know i can use redirects to fix the new path), but i would prefere to keep the structure intact, if possible. 

Any suggestion will be appreciated!
thank you

**Comment by wkdhkr** [comment]
You can simply use `[...slug].js`.

https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes

**Reply by offwork** [reply]
`[...slug].js` a terrible response. Already `optional catch-all` is done with the use of double square brackets. And unfortunately **nextjs** documentation is in a pathetic state.

**Reply by sachingvit** [reply]
While I am trying to create dynamic routing for route pattern

```/category1?/category2?/category3?/category4?/c/categoryId```

Here category1 to category4 are optional in route but after that ```/c/categoryId``` is fix.

In order to solution I had tried [optional catch-all-segments](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#optional-catch-all-segments)
```
- [[...category]]
------- c
-----------[categoryId]
---------------- page.tsx
```
But getting error that ```[Error: Invalid segment Dynamic("categoryId"), catch all segment must be the last segment (segments: [OptionalCatchAll("category")])```


https://stackoverflow.com/questions/78427872/nextjs-14-optional-catch-all-segments-dynamic-routing-not-working-for-optional
